src/Neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -220,7 +240,9 @@ public Header GetHeader(DataCache snapshot, uint index)
         /// <returns>The <see cref="TransactionState"/> with the specified hash.</returns>
         public TransactionState GetTransactionState(DataCache snapshot, UInt256 hash)
         {
-            return snapshot.TryGet(CreateStorageKey(Prefix_Transaction).Add(hash))?.GetInteroperable<TransactionState>();
+            var state = snapshot.TryGet(CreateStorageKey(Prefix_Transaction).Add(hash))?.GetInteroperable<TransactionState>();
+            if (state is not null && state.Trimmed) return null;
```

Comment:
> the state of a `Trimmed` tx stored on the Blockchain is `null`

No. The `TransactionState` can be null if there's no such key in the storage, i.e. there's no transaction or conflicting dummy transaction with provided hash.

> It is because it is just the hash? There is not state in fact?

There is a `TransactionState`, but it's trimmed (with empty `Transaction`). This means that there's some on-chain transaction that has `Conflicts` attribute with provided hash. `GetTransactionState` shouldn't return such trimmed `TransactionState`.

Reply:
> Maybe `trimmed` is vague and we could pick another nomenclature.

Which one would you like?